src/Components/Selector/Selector.js

```diff
@@ -0,0 +1,42 @@
+import React, { useState, useEffect } from 'react';
+import './Selector.css';
+import { getStates, getCities } from '../../apiCalls';
+
+const Selector = ({getLocationData}) => {
+
+  const [availableStates, setAvailableStates] = useState([]);
+  const [state, setState] = useState('');
+  const [availableCities, setAvailableCities] = useState([]);
+  const [city, setCity] = useState('');
+
+  useEffect(async () => {
+    setAvailableStates(await getStates())
+  }, [])
+
+  useEffect(async () => {
+    if (state) {
+      setAvailableCities(await getCities(state))
+    }
+  }, [state])
+
+  useEffect(async () => {
+    if (city) {
+      getLocationData(state, city)
+    }
+  }, [city])
+
+  return (
+    <form>
+      <select onChange={event => setState(event.target.value)}>
+        <option value=''>Select State</option>
+        {availableStates.map(state => <option value={state}>{state}</option>)}
```

Comment:
I think it would be best to address this now; the rubric does want logic kept out of return statements. Since this is a functional component, we can move this to a variable outside out of the return, and call that variable here.

Reply:
Thanks for suggesting this, I have move out the logic and calling the variable only now.